templates/site/js/dataproduct-form.js

```diff
@@ -863,18 +938,29 @@ function submitForm(){
         }
     }
 
-    /*Add date input to submitObject*/
-    if(dateMetadataFields[0].value == "" && dateMetadataFields[1].value == ""){
-        for(dateMetadata of dateMetadataFields){
-            metadataObject[dateMetadata.id] = null;
-        }
-    }
-    else{
-        for(dateMetadata of dateMetadataFields){
-            metadataObject[dateMetadata.id] = dateMetadata.value;
+    /*Add Metadata Other input to metadataObject*/
+    if(otherMetadataInputFields.length > 0){
+        for (otherInput of otherMetadataInputFields) {
+            var otherInputIDArray = otherInput.id.split("_");
+            geometryType = otherInputIDArray[0];
+            geometryContainer = document.getElementById(geometryType + "_box");
+
+            if (otherInput.value != "") {
```

Comment:
If a key with no value is invalid (and won't be sent to the backend) then it should be marked invalid on the form.
Note that you probably want to treat a row with no key and no value differently from a form with no key and a value